modules/lfs/http_client.go

```diff
@@ -113,6 +115,7 @@ func (c *HTTPClient) Upload(ctx context.Context, objects []Pointer, callback Upl
 	return c.performOperation(ctx, objects, nil, callback)
 }
 
+// performOperation takes a slice of LFS object pointers, batches them, and performs the upload/download operations concurrently in each batch
```

Comment:
One more thing, I do not think "batch size" means "concurrency limit" in this case. If I understand correctly:

* "batch size" means how many objects to be processed in one time.
* "concurrency limit" means how many concurrent connections to the LFS server.

Maybe "batch size" could be 100 or 1000 without causing problem, but "concurrency limit" should be much smaller, eg: 5 or 10. A lot of connections to a LFS server might trigger their rate-limit protection, or be considered as somewhat DoS?

Reply:
I'm always in favor of making things more configurable, however as an administrator I think most of the time you would want `[lfs_client].BATCH_SIZE` = (the number of concurrent threads for upload/download within a batch). I'm happy to add a new config though, what should we call it and what should its default be? `[lfs_client].BATCH_MAX_THREADS` ?

Reply:
Maybe `[lfs_client].MAX_OPERATION_CONCURRENCY` ? (since there is no "thread" concept in Golang)

Reply:
Fair, how about `[lfs_client].BATCH_OPERATION_CONCURRENCY`, as I feel there may be other single-threaded operations still in this code that we might want configs for then, and this is specific to lfs client batch operations? I updated this branch with that config, and defaulted it to what `[lfs_client].BATCH_SIZE` if unset or <1

Reply:
We need to match the default git-lfs's behavior:

https://github.com/git-lfs/git-lfs/blob/4cca3f8e7d77c5b86cdfe34c2e6bba2126e3d5cd/docs/man/git-lfs-config.5.ronn#L25-L33

```
lfs.concurrenttransfers

The number of concurrent uploads/downloads. Default 3.
```